code/modules/arcade/mob_hunt/mob_avatar.dm

```diff
@@ -12,17 +11,21 @@
 	var/list/clients_encountered = list()		//tracks who has already interacted with us, so they can't attempt a second capture
 	var/image/avatar
 
-/obj/effect/nanomob/New(loc, datum/mob_hunt/new_info)
-	..()
+/obj/effect/nanomob/Initialize(mapload, datum/mob_hunt/new_info)
+	. = ..()
 	if(!new_info)
-		qdel(src)
-		return
+		return INITIALIZE_HINT_QDEL
 	mob_info = new_info
 	update_self()
 	forceMove(mob_info.spawn_point)
 	if(!mob_info.is_trap)
 		addtimer(CALLBACK(src, .proc/despawn), mob_info.lifetime)
 
+/obj/effect/nanomob/Destroy()
+	SSmob_hunt.trap_spawns -= src
+	SSmob_hunt.normal_spawns -= src
```

Comment:
```suggestion
	SSmob_hunt.normal_spawns -= src
	mob_info = null
	clients_encountered.Cut()
	qdel(avatar)
```
Not sure if you need to delete the avatar btw

Reply:
![image](https://user-images.githubusercontent.com/57483089/131232628-63d3ed25-9a51-4a73-9812-a6de490cd14f.png)
None of those seem to be causing hard deletes so I guess I must've just overlooked it. Still good to remove anyway though. 👍

Reply:
These will most likely not affect the nanomob itself but they might affect the mob_hunt object stored in `mob_info` for instance